File: Code Generation/responses/cpp/claude/easy/minimum-cost-to-reach-every-position.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<int> minCosts(vector<int> cost) {
        int n = cost.size();
        vector<int> answer(n);
        int minSoFar = cost[0];
        for (int i = 0; i < n; i++) {
            minSoFar = min(minSoFar, cost[i]);
            answer[i] = minSoFar;
        }
        return answer;
    }
};

File: Code Generation/responses/python/claude/easy/minimum-cost-to-reach-every-position.py
from typing import List

class Solution:
    def minCosts(self, cost: List[int]) -> List[int]:
        n = len(cost)
        answer = []
        min_so_far = float('inf')
        for i in range(n):
            min_so_far = min(min_so_far, cost[i])
            answer.append(min_so_far)
        return answer